### PR TITLE
Merge similar GitHub Actions workflow files #2083

### LINKED ISSUE #2083: Merge similar GitHub Actions workflow files
## Refactor

Workflow files are highly similar, which can be refactored by matrix and merged into one workflow.
https://github.com/apache/shardingsphere-elasticjob/tree/master/.github/workflows

### PR DESCRIPTION
**Pull request by m1ngyuan**
Fixes #2083.

Changes proposed in this pull request:
- Merge similar GitHub Actions workflow files

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -705,6 +705,12 @@
                 </pluginManagement>
             </build>
         </profile>
+        <profile>
```
Comment (TeslaCN) [inline pom.xml]: What's the different with `-Dmaven.javadoc.skip=true`?

**Review by TeslaCN — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (m1ngyuan) [reply pom.xml]: -Dmaven.javadoc.skip=true on windows will be fail

Comment (TeslaCN) [reply pom.xml]: > -Dmaven.javadoc.skip=true on windows will be fail

Why? Could you check it again? I think it doesn't make sense.

Comment (m1ngyuan) [reply pom.xml]: Thank you for your advice,I found a solution to the problem,Please trigger GitHub workflow again

**Review by TeslaCN — APPROVED** [review]
Thank you!

**Pull request merged**